driver/Main.hs

```diff
@@ -21,12 +24,39 @@ header = unlines [
   , ""
   ]
 
+data Option = Help
+  deriving (Show, Eq)
+
+options :: [OptDescr Option]
+options = [
+    Option ['h'] ["help"] (NoArg Help) "show help and exit"
+  ]
+
+usage :: String
+usage = usageInfo ("hpack " ++ showVersion version) options
+
 main :: IO ()
 main = do
+  (opts, args, errs) <- fmap (getOpt Permute options) getArgs
+  unless (null errs) (printUsage errs)
+  unless (null args) (printUsage ["hpack does not accept non-option arguments\n"])
```

Comment:
Right. There are also extra empty lines between error messages. I think it's reasonable to remove them too.